services/admin/handlers/updatePackageId.js

```diff
@@ -0,0 +1,146 @@
+import AWS from "aws-sdk";
+
+const dynamoDb = new AWS.DynamoDB.DocumentClient(
+  process.env.IS_OFFLINE
+    ? {
+        endpoint: "http://localhost:8000",
+      }
+    : {}
+);
+const oneMacTableName = process.env.oneMacTableName;
+
+function validateEvent(event) {
+  //validate required input params
+  let missingParams = "";
+
+  if (!event.fromPackageId) {
+    missingParams += " fromPackageId ";
+  }
+  if (!event.toPackageId) {
+    missingParams += " toPackageId ";
+  }
+  if (!event.prependAdditionalInfo) {
+    missingParams += " prependAdditionalInfo ";
+  }
+  if (missingParams.trim().length != 0) {
+    throw new Error("Missing event parameters - " + missingParams);
+  }
+
+  console.log("event passed validation");
+}
+
+async function getOneMacEventRecords(fromPackageId) {
+  const params = {
+    TableName: oneMacTableName,
+    KeyConditionExpression: "pk = :pkValue and begins_with(sk, :skValue)",
+    ExpressionAttributeValues: {
+      ":pkValue": fromPackageId,
+      ":skValue": "OneMAC#",
+    },
+  };
+
+  console.log("right  before query", params);
+  const queryResponse = await dynamoDb.query(params).promise();
+  console.log("OneMAC events found for: ", fromPackageId);
+
+  return queryResponse.Items;
+}
+
+function formatPutRequests(toPackageId, addlInfo, records) {
+  const putRequests = records.map((item) => {
+    const auditArray = item.auditArray
+      ? [...item.auditArray, addlInfo]
```

Comment:
SEA Tool's pattern is to prepend the most recent information to the front of whatever.... probably should have this the same direction...

Reply:
updated this order in latest commit